go/tasks/plugins/presto/execution_state.go

```diff
@@ -0,0 +1,528 @@
+package presto
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"strings"
+
+	"github.com/lyft/flyteplugins/go/tasks/plugins/presto/client"
+
+	"time"
+
+	"github.com/lyft/flytestdlib/cache"
+
+	idlCore "github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/lyft/flyteidl/gen/pb-go/flyteidl/plugins"
+
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/utils"
+	"github.com/lyft/flyteplugins/go/tasks/plugins/presto/config"
+
+	"github.com/lyft/flyteplugins/go/tasks/errors"
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/core"
+	"github.com/lyft/flytestdlib/logger"
+
+	"github.com/lyft/flyteplugins/go/tasks/plugins/svc"
+)
+
+type ExecutionPhase int
+
+const (
+	PhaseNotStarted ExecutionPhase = iota
+	PhaseQueued                    // resource manager token gotten
+	PhaseSubmitted                 // Sent off to Presto
+	PhaseQuerySucceeded
+	PhaseQueryFailed
+)
+
+func (p ExecutionPhase) String() string {
+	switch p {
+	case PhaseNotStarted:
+		return "PhaseNotStarted"
+	case PhaseQueued:
+		return "PhaseQueued"
+	case PhaseSubmitted:
+		return "PhaseSubmitted"
+	case PhaseQuerySucceeded:
+		return "PhaseQuerySucceeded"
+	case PhaseQueryFailed:
+		return "PhaseQueryFailed"
+	}
+	return "Bad Presto execution phase"
+}
+
+type ExecutionState struct {
+	Phase ExecutionPhase
+
+	// This will store the command ID from Presto
+	CommandID string `json:"command_id,omitempty"`
```

Comment:
We are trying to standardize on camel casing for json tags.
```suggestion
	CommandID string `json:"commandId,omitempty"`
```

Reply:
Hmm, if this is the case, then it appears it would be inconsistent with what you guys have for the Hive  plugin: https://github.com/lyft/flyteplugins/blob/master/go/tasks/plugins/hive/execution_state.go#L54
Do you guys prefer camelCase or snake_case?

Reply:
camel I'm pretty sure.